documentation/src/docs/streaming.md

```diff
@@ -0,0 +1,194 @@
+
+## FlinkPravegaWriter
+A Pravega Stream may be used as a data sink within a Flink program using an instance of `io.pravega.connectors.flink.FlinkPravegaWriter`. Add an instance of the writer to the dataflow program using the method [`DataStream::addSink`](https://ci.apache.org/projects/flink/flink-docs-stable/api/java/org/apache/flink/streaming/api/datastream/DataStream.html#addSink-org.apache.flink.streaming.api.functions.sink.SinkFunction-).
+
+### Example
+```Java
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+// Define the Pravega configuration
+PravegaConfig config = PravegaConfig.fromParams(params);
+
+// Define the event serializer
+SerializationSchema<MyClass> serializer = ...
+
+// Define the event router for selecting the Routing Key
+PravegaEventRouter<MyClass> router = ...
+
+// Define the sink function
+FlinkPravegaWriter<MyClass> pravegaSink = FlinkPravegaWriter.<MyClass>builder()
+   .forStream(...)
+   .withPravegaConfig(config)
+   .withSerializationSchema(serializer)
+   .withEventRouter(router)
+   .withWriterMode(EXACTLY_ONCE)
+   .build();
+
+DataStream<MyClass> stream = ...
+stream.addSink(pravegaSink);
+```
+### Parameters
+
+A builder API is provided to construct an instance of `FlinkPravegaWriter`. See the table below for a summary of builder properties.  Note that the builder accepts an instance of `PravegaConfig` for common configuration properties.  See the [configurations](configurations.md) page for more information.
+
+|Method                |Description|
+|----------------------|-----------------------------------------------------------------------|
+|`withPravegaConfig`|The Pravega client configuration, which includes connection info, security info, and a default scope.|
+|`forStream`|The stream to be written to.|
+|`withWriterMode`|The writer mode to provide _Best-effort, _At-least-once_, or _Exactly-once_ guarantees.|
+|`withTxnLeaseRenewalPeriod`|The Transaction lease renewal period that supports the _Exactly-once_ writer mode.|
+|`withSerializationSchema`|The serialization schema which describes how to turn events into byte messages.|
+|`withEventRouter`|The router function which determines the Routing Key for a given event.|
+|`enableMetrics`|true or false to enable/disable reporting Pravega metrics. Metrics is enabled by default.|
+
+### Parallelism
+`FlinkPravegaWriter` supports parallelization. Use the `setParallelism` method to configure the number of parallel instances to execute.
+
+### Event Routing
+Every event written to a Pravega Stream has an associated Routing Key.  The Routing Key is the basis for event ordering.  See the [Pravega Concepts](http://pravega.io/docs/latest/pravega-concepts/#events) for details.
+
+When constructing the `FlinkPravegaWriter`, please provide an implementation of `io.pravega.connectors.flink.PravegaEventRouter` which will guarantee the event ordering. In Pravega, events are guaranteed to be ordered at the segment level.

```

Comment:
Don't we have to say how the application is supposed to extract the key?

Reply:
Added a code snippet to show an example